modules/_internal/runner/main.tf

```diff
@@ -19,13 +23,7 @@ locals {
     type         = "LINUX_CONTAINER"
   }
 
-  # hack: <https://github.com/hashicorp/terraform/issues/12453#issuecomment-378033384>
-  environment_variables = {
-    empty     = "${map()}"
-    not_empty = "${map("environment_variable", var.environment_variables)}"
-  }
-
-  environment = "${merge(local.default_environment, var.environment, local.environment_variables[length(var.environment_variables) > 0 ? "not_empty" : "empty"])}"
+  environment = merge(local.default_environment, var.environment)
```

Comment:
This is so much cleaner than the TF 0.11.x hack

Reply:
The super cool part is that the upgrade command `terraform 0.12upgrade` actually created the dynamic blocks for the `environment` and `environment_variable`, so I could see pretty well how that worked. Then with a little studying, I realized the hack was no longer needed at all!